src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -528,13 +528,14 @@ double HMM<Distribution>::Predict(const arma::mat& dataSeq,
     for (size_t j = 0; j < logTransition.n_rows; j++)
     {
       arma::vec prob = logStateProb.col(t - 1) + logTransition.row(j).t();
-      logStateProb(j, t) = prob.max(index) + logProbs(t, j);
+      index = prob.index_max();
+      logStateProb(j, t) = index + logProbs(t, j);
```

Comment:
```suggestion
      logStateProb(j, t) = prob[index] + logProbs(t, j);
```
I think that this is the cause of the test failure.

Reply:
Thank you!  The edit, cmake, test cycle is soooooo looooong I just got to reaffirm failures here.  And apparently I suck because I also have

```sh
-------------------------------------------------------------------------------
/home/edd/git/mlpack/src/mlpack/tests/perceptron_test.cpp:107
...............................................................................
                                                                                                         
/home/edd/git/mlpack/src/mlpack/tests/perceptron_test.cpp:125: FAILED:
  REQUIRE( predictedLabels(2) == 1 )             
with expansion:
  0 == 1

-------------------------------------------------------------------------------
Or
-------------------------------------------------------------------------------
/home/edd/git/mlpack/src/mlpack/tests/perceptron_test.cpp:143
...............................................................................
                                                    
/home/edd/git/mlpack/src/mlpack/tests/perceptron_test.cpp:160: FAILED:
  REQUIRE( predictedLabels(0, 0) == 1 )          
with expansion:
  0 == 1

```

Reply:
That would appear to be similar. I think I want

```
    maxIndex = tempLabelMat.index_max();
    predictedLabels(i) = tempLabelMat[maxIndex];     // ie index into variable here too
```

Thumbs up or down?